### DISCUSSION TITLE
Replicating data from MONOLIX

### DISCUSSION THREAD

**Discussion by zeref221** [metrumresearchgroup/mrgsolve#1004] [Q&A]
Hello,

I have couple of questions:

1. I am trying to reproduce a monolix code in mrgsolve. Is there some quick way of doing this that I might be unaware of?

2. I have simulated individual dosing regiments but I am finding it difficult to simulate for a large population. Can you suggest some reference to go about adding random effects, eta's and covariates? 

3. Lastly can you list out things that are out of scope for mrgsolve which can be done in monolix?

I am fairly new to mrgsolve, I have watched previous youtube videos explaining sensitivity analysis and adding covariates but still a little confused.

I'd be happy to share any snippets of code if required.

Thanks in advance!

**Comment by kylebaron** [comment]
Hi @zeref221 - 

Thanks for reaching out in the Discussion forum. I don't have experience using Monolix, but I imagine the behavior is fairly similar to NONMEM and mrgsolve is similar to that. So assuming the differences in implementation are acceptable, I'm guessing that with a standard model and data set the results will be similar. 

When I'm trying to replicate NONMEM data and I'm not sure about the model and data, I usually do a quick check to make sure they are doing the same or if not identical, substantially similar. 

I don't have a list of mrgsolve scope as compared to Monolix, but I have some tests against NONMEM which might be helpful:
https://github.com/mrgsolve/nmtests/blob/master/dosing/dosing-vignette.md

For examples, you could try this model depot I put together.  You can find population models here:

https://github.com/mrgsolve/depot/blob/master/pkg/inst/models/moxi.cpp
https://github.com/mrgsolve/depot/blob/master/pkg/inst/models/pembro.cpp
https://github.com/mrgsolve/depot/blob/master/pkg/inst/models/sunit.cpp
https://github.com/mrgsolve/depot/blob/master/pkg/inst/models/secukinumab.cpp
https://github.com/mrgsolve/depot/blob/master/pkg/inst/models/meropenem.cpp
https://github.com/mrgsolve/depot/blob/master/pkg/inst/models/azithro.cpp
https://github.com/metrumresearchgroup/mrgsolve-summer-2020/blob/master/model/palivizumab.cpp

These should be referenced (in the code) and could provided examples to follow.  

There is also model specification syntax here:
https://mrgsolve.org/user_guide/model-specification.html

And another example here
https://mrgsolve.org/user_guide/model-specification.html#population-pk-model-with-covariates-and-iov

I hope this helps! Please let me know if you have specific questions once you get into it.
Kyle
